src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/CallInstruction.cs

```diff
@@ -418,15 +479,27 @@ public sealed override int Run(InterpretedFrame frame)
                     {
                         args[i] = frame.Data[first + i + 1];
                     }
+
+                    instance = frame.Data[first];
+
+                    LightLambda targetLambda;
+                    if (TryGetLightLambdaTarget(instance, out targetLambda))
+                    {
+                        // no need to Invoke, just interpret the lambda body
+                        ret = InterpretLambdaInvoke(targetLambda, args);
+                        goto invoked;
```

Comment:
good point. This went through several changes. It looks like the final variant could use else